### ISSUE TITLE
rpm testsuite test 236 fails with file 5.40

### ISSUE THREAD

**Issue by wally-mageia** [rpm-software-management/rpm#1616]
Rpm build fails with test suite error after updating file to 5.40 in Mageia Cauldron.

```
## ------------------------ ##
## Summary of the failures. ##
## ------------------------ ##
Failed tests:
rpm 4.16.1.3 test suite test groups:

 NUM: FILE-NAME:LINE     TEST-GROUP-NAME
      KEYWORDS

 236: rpmsigdig.at:144   rpmkeys -Kv <unsigned> 2
      rpmkeys digest

## ---------------------- ##
## Detailed failed tests. ##
## ---------------------- ##

#                             -*- compilation -*-
236. rpmsigdig.at:144: testing rpmkeys -Kv <unsigned> 2 ...
./rpmsigdig.at:146:


if ! [ -d testing/ ]; then
    cp -aP "${RPMTEST}" .
    chmod -R u+w testing/
    mkdir -p testing/build
    ln -s ../data/SOURCES testing/build/
fi
export RPMTEST="${PWD}/testing"
export TOPDIR="${RPMTEST}/build"
export HOME="${RPMTEST}"

rm -rf "${RPMTEST}"`rpm --eval '%_dbpath'`/*
runroot rpm --initdb


runroot rpmbuild -bb --quiet \
	--define "%optflags -O2 -g" \
	--define "%_target_platform noarch-linux" \
	--define "%_binary_payload w.ufdio" \
	--define "%_buildhost localhost" \
	--define "%use_source_date_epoch_as_buildtime 1" \
	--define "%source_date_epoch_from_changelog 1" \
	--define "%clamp_mtime_to_source_date_epoch 1" \
	/data/SPECS/attrtest.spec
for v in SHA256HEADER SHA1HEADER SIGMD5 PAYLOADDIGEST PAYLOADDIGESTALT; do
    runroot rpm -q --qf "${v}: %{${v}}\n" /build/RPMS/noarch/attrtest-1.0-1.noarch.rpm
done
runroot rpmkeys -Kv /build/RPMS/noarch/attrtest-1.0-1.noarch.rpm

--- -	2021-03-31 19:04:59.904829145 +0000
+++ /builddir/build/BUILD/rpm-4.16.1.3/tests/rpmtests.dir/at-groups/236/stdout	2021-03-31 19:04:59.901706164 +0000
@@ -1,6 +1,6 @@
-SHA256HEADER: 3959e329ad3cad72f47db0d55539d7adcd78df6f6989edbbcca80f8b1990e032
-SHA1HEADER: 854197e167029333a110e5c879730be89cd118b5
-SIGMD5: 237a75a9367feae4c21044882e7acae7
+SHA256HEADER: fdf728f50cb985830bd7117f5ea42655a085b9f512fc714bbeea2097048adcf2
+SHA1HEADER: 62dd0d50bf88a7ceb4b2e4026618f55f3b1135ef
+SIGMD5: 6181cb2e9db116f345beceb914124808
 PAYLOADDIGEST: 749d8980cc5889419da8cdbe9a5b3292742af8a227db3635f84966481b7612a8
 PAYLOADDIGESTALT: 749d8980cc5889419da8cdbe9a5b3292742af8a227db3635f84966481b7612a8
 /build/RPMS/noarch/attrtest-1.0-1.noarch.rpm:
236. rpmsigdig.at:144: 236. rpmkeys -Kv <unsigned> 2 (rpmsigdig.at:144): FAILED (rpmsigdig.at:146)
```
Failing test succeeds if I revert file/file@d604c5d in file. File upstream's [bug report 180](https://bugs.astron.com/view.php?id=180) related to commit causing the failure.

**Comment by pmatilai** [comment]
Right, it's just fileclass string change causing the checksums not match expectations. The files in question are created like this:
```
for x in a b c d e f g h i j; do
    mkdir -p $RPM_BUILD_ROOT/${x}
    mkdir -p $RPM_BUILD_ROOT/${x}/dir
    echo "${x}" > $RPM_BUILD_ROOT/${x}/file
    chmod 700 $RPM_BUILD_ROOT/${x}/dir
    chmod 400 $RPM_BUILD_ROOT/${x}/file
done

```

Ie the regular files contain just one ASCII letter, and so it'll certainly report something else if three letters are required for ASCII classification. Easy enough to work around by adding more gunk to the files of course.

**Comment by pmatilai** [comment]
PR #1624 should fix, but I can't easily test with file 5.40. So if you can confirm, that'd be appreciated.
